### DISCUSSION TITLE
Using versioning=semver-coerced sometimes fails to remove leading v

### DISCUSSION THREAD

**Discussion by MAGICCC** [renovatebot/renovate#24932] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I have a case when using semver-coerced as versioning in a Dockerfile, the PR created by renovate is wrong.
Normally the leading 'v' should be removed from the version.

Is there something wrong what I am missing since all the tags from upstream are with leading 'v' and it was working fine before
You can check it @ https://github.com/MAGICCC/improved-octo-enigma/pull/3/files

### Relevant debug logs

<details><summary>Logs</summary>

```
{
  "baseBranch": "master"
  "config": {
    "regex": [
      {
        "deps": [
          {
            "depName": "php/pecl-mail-mailparse",
            "currentValue": "3.1.4",
            "datasource": "github-tags",
            "versioning": "semver-coerced",
            "replaceString": "# renovate: datasource=github-tags depName=php/pecl-mail-mailparse versioning=semver-coerced\nARG MAILPARSE_PECL_VERSION=3.1.4\n",
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "v3.1.6",
                "newValue": "v3.1.6",
                "releaseTimestamp": "2023-08-22T06:11:12.000Z",
                "newMajor": 3,
                "newMinor": 1,
                "updateType": "patch",
                "branchName": "renovate/php-pecl-mail-mailparse-3.x"
              }
            ],
            "packageName": "php/pecl-mail-mailparse",
            "warnings": [],
            "sourceUrl": "https://github.com/php/pecl-mail-mailparse",
            "registryUrl": "https://github.com",
            "currentVersion": "3.1.4",
            "isSingleVersion": true,
            "fixedVersion": "3.1.4"
          }
        ],
        "matchStrings": [
          "#\\srenovate:\\sdatasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\s(ENV|ARG) .*?_VERSION=(?<currentValue>.*)\\s"
        ],
        "packageFile": "Dockerfile"
      }
    ]
  }
}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
You should use `extractVersion` to remove the v prefix
